src/dev/build/tasks/nodejs_modules/code_parser/visitors.js

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export function dependenciesVisitorsGenerator(dependenciesAcc) {
+  return (() => {
+    // This was built based on two main tools: an ast explorer and the
+    // main docs for the Esprima
+    //
+    // https://astexplorer.net
+    // https://esprima.readthedocs.io/en/latest/syntax-tree-format.html
+    return {
+      // Visitors to traverse and found dependencies
+      // raw values on require + require.resolve
+      CallExpression: ({ node }) => {
+        // AST check for require expressions
+        const isRequire = (node) => {
+          return node.callee && node.callee.type === 'Identifier' && node.callee.name === 'require';
+        };
+
+        // AST check for require.resolve expressions
+        const isRequireResolve = (node) => {
+          return node.callee && node.callee.type === 'MemberExpression' && node.callee.object
+            && node.callee.object.type === 'Identifier' && node.callee.object.name === 'require'
+            && node.callee.property && node.callee.property.type === 'Identifier'
+            && node.callee.property.name === 'resolve';
```

Comment:
You might be able to use lodash's [_.matches](https://lodash.com/docs/4.17.10#matches) function to make this easier to read.

Reply:
good idea. updated